File: backend-javascript/03-backend-javascript-postgresql-old/src/features/person/person-repository.js
import PersonPgRepository from './person-repository-pg.js';
import PersonMockRepository from './person-repository-mock.js';

class PersonRepository {
  constructor(useDatabase) {
    this.repository = useDatabase
      ? new PersonPgRepository()
      : new PersonMockRepository();
  }

  async getItems() {
    return this.repository.getItems();
  }

  async getItemById(id) {
    return this.repository.getItemById(id);
  }

  async createItem(data) {
    return this.repository.createItem(data);
  }

  async updateItem(id, data) {
    return this.repository.updateItem(id, data);
  }

  async deleteItem(id) {
    return this.repository.deleteItem(id);
  }
}

export default PersonRepository;

File: backend-java/03-backend-java-postgresql/src/main/java/com/ganatan/modules/person/PersonRepository.java
package com.ganatan.modules.person;

import java.util.List;

import com.ganatan.mocks.PersonRepositoryMock;

public class PersonRepository {

    private final Object repository;

    public PersonRepository(boolean useDatabase) {
        if (useDatabase) {
            this.repository = new PersonRepositoryMock();
        } else {
            this.repository = new PersonRepositoryMock();
        }
    }

    public List<Person> getItems() {
        return ((PersonRepositoryMock) repository).getItems();
    }
}

